test/extended/util/oauthserver/oauthserver.go

```diff
@@ -0,0 +1,546 @@
+package oauthserver
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"math/rand"
+	"path"
+	"reflect"
+	"time"
+
+	"github.com/RangelReale/osincli"
+
+	corev1 "k8s.io/api/core/v1"
+	rbacv1 "k8s.io/api/rbac/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/util/wait"
+	restclient "k8s.io/client-go/rest"
+
+	configv1 "github.com/openshift/api/config/v1"
+	legacyconfigv1 "github.com/openshift/api/legacyconfig/v1"
+	oauthv1 "github.com/openshift/api/oauth/v1"
+	osinv1 "github.com/openshift/api/osin/v1"
+	configclient "github.com/openshift/client-go/config/clientset/versioned"
+	"github.com/openshift/library-go/pkg/config/helpers"
+	"github.com/openshift/library-go/pkg/crypto"
+	"github.com/openshift/oc/pkg/helpers/tokencmd"
+	"github.com/openshift/origin/test/extended/testdata"
+	exutil "github.com/openshift/origin/test/extended/util"
+)
+
+const (
+	serviceURLFmt = "https://test-oauth-svc.%s.svc" // fill in the namespace
+
+	servingCertDirPath  = "/var/config/system/secrets/serving-cert"
+	servingCertPathCert = "/var/config/system/secrets/serving-cert/tls.crt"
+	servingCertPathKey  = "/var/config/system/secrets/serving-cert/tls.key"
+
+	routerCertsDirPath = "/var/config/system/secrets/router-certs"
+
+	sessionSecretDirPath = "/var/config/system/secrets/session-secret"
+	sessionSecretPath    = "/var/config/system/secrets/session-secret/session"
+
+	oauthConfigPath  = "/var/config/system/configmaps/oauth-config"
+	serviceCADirPath = "/var/config/system/configmaps/service-ca"
+
+	configObjectsDir = "/var/oauth/configobjects/"
+
+	RouteName = "test-oauth-route"
+	SAName    = "e2e-oauth"
+)
+
+var (
+	serviceCAPath = "/var/config/system/configmaps/service-ca/service-ca.crt" // has to be var so that we can use its address
+
+	defaultProcMount         = corev1.DefaultProcMount
+	volumesDefaultMode int32 = 420
+)
+
+// DeployOAuthServer - deployes an instance of an OpenShift OAuth server
+// very simplified for now
+// returns OAuth server url, cleanup function, error
+func DeployOAuthServer(oc *exutil.CLI, idps []osinv1.IdentityProvider, configMaps []corev1.ConfigMap, secrets []corev1.Secret) (*tokencmd.RequestTokenOptions, func(), error) {
```

Comment:
if you end up having to expand this,  would consider creating chainable functions just for producing a configuration.  This is ok to start, but think `DefaultConfig().WithIDP().SetMyFunc().ToConfigOrDie()`

Reply:
> prefer your own struct with `NewFromCLI` constructor. Consider what would happen if you were disconnected from kube ginkgo.

This doesn't haven't to happen before merge.